src/java.base/share/native/libjava/ClassLoader.c

```diff
@@ -99,7 +99,14 @@ Java_java_lang_ClassLoader_defineClass1(JNIEnv *env,
         return 0;
     }
 
+    // On malloc(0), implementators of malloc(3) have the choice to return either
```

Comment:
It is confusing to mix `malloc(0)`, where you are passing an argument zero to malloc, with `malloc(3)` which actually means the definition of malloc as per the man page in section 3.

Given this is only an issue on AIX the comment can simply say:

`// On AIX malloc(0) returns NULL which looks like an out-of-memory condition; so adjust it to malloc(1)`.

Reply:
I would omit the bug number reference, they get stale and do not age well, cluttering up the source.
Git blame can be used to find the origin of the comment if needed.